### DISCUSSION TITLE
Looking for co-maintainers

### DISCUSSION THREAD

**Discussion by jesper-raemaekers** [jesper-raemaekers/python-polarion#148] [Announcements]
Hi All,

Recently I've switched jobs and this means I no longer work with Polarion on a daily basis. I've noticed by response times slowing because of this. For me, the main thing I dislike now is not being able to approve PR fast enough, as I do encourage anyone to contribute, but having to wait while doing so can be demotivating. So I think it is time to ask for help.
I'm looking for one (or more) people to help with this project. I'll include you in the GitHub and give you access where needed (DevOps, testing VPS, polarion docker, whatever). As I'm also new to this, we'll figure it out.

Is anybody willing to help?

Now just tagging some recently active people; @LionnelC @TristanFAURE  @therealevahill @fieschkon

**Comment by nunobrum** [comment]
Hello @jesper-raemaekers ,

I've been using your library and I've made a few edits to it, which I think would benefit the whole community. 
For example, I've implemented a class for managing the test steps, as I noticed that the current approach generated a lot of commits to the database.
This was before the implementation of the postpone_save feature, which, also needed a few corrections in order to cover some situations.
I was a bit demotivated to make pull-requests exactly for the reason you mention on your email. Now, seeing your message, I understand your reasons, and volunteering for help.
I can start by making a pull-request with these changes and some other bugfixes I've made on my fork.
There is however a topic that I would like to introduce here.  Polarion has been actively developing the REST API. This is an alternative to the WebServices. I don't know what is your experience with WebServices, but I often run into issues on the underlying zeep library.
Unrecognized types when operating with EnumIds, unsupported characters, and so on.
I guess this is one of the reasons that lead Siemens to develop and now advertise the REST API.  
I was considering making some tests with REST API and if things work well, even have some parts of the implementation using this new interface. Provided any of this is even possible, would it be something that would be welcomed here, or, is it better to keep it as a separate development?

Best Regards,
Nuno Brum

**Reply by jesper-raemaekers** [reply]
Hi @nunobrum,

I welcome your help and any pr to fix these issues. If you'd like I can add you as maintainer here. Now let's jump to the next question, which is more interesting in my opinion.

Funny you mention zeep, because this package was created because working with it was, well, not great. 

I'm aware of the rest api and even had some test releases with this functionality from Siemens. I'm unsure on the release status though? 
I think a new repo, and new package would be best because this package has a fundamental circular dependency issue and trying to isolate the api is annoying because rest is way more structured. There is an issue on the structure in but I cannot find it on mobile currently.

I'm motivated to get the new stuff started, setup the structure and test environment. But I don't have enough time to pull that alone.

**Reply by nunobrum** [reply]
Oh! Wow. Somehow I've missed your reply and only seeing it now after a long due cleanup.
I am happy to see that my pull-requests were accepted. Congratulations for your project.  Awesome work.

**Comment by TristanFAURE** [comment]
Hello
about the REST API do we know if Siemens could provide an "official" library (python, java...) in the future ?

About contributing I need to check with @LionnelC how much we will use this in our industrial context and if yes that could be interesting to be more involved

**Comment by therealevahill** [comment]
Hi Jesper,

sorry for the long delay. Icannot use our company account to use github, we use this eva-hill account to access the world wide availabe github. So until now Ilooked sometimes it to this account, but this is changing now. So that this account is used every day.
I am using your library daily.
If there is a RestApi available, I would prefere that as far as it provides the same functionality.

**Comment by mfieschko** [comment]
Hey Jesper,

I was waiting to see if there would be further discussion on this, but it looks there haven't been too many updates. I use Polarion daily as a part of the engineering process where I work, and I use this package to develop various internal tools for it.

First, I'd be happy to help look over PR's. My only reservation (and the reason I've held off answering for so long) is that I'm not super familiar with the inner workings of Zeep or SOAP/WSDL in general. That being said, I can still do my best to help maintain PR's.

@nunobrum @jesper-raemaekers with regards to REST, I propose adding that functionality to the current Polarion client object. I've noticed that the different APIs have some unique commands that aren't on the other. Perhaps something like `client.REST.<somecommand>(<someparameters>)`? That way it's all under one installation package and one client so you don't have to initialize authentication with several packages, and you can still use all commands. Not sure how much refactoring that would take. The issue I see here is that (to my knowledge) you can't authenticate REST requests with username and password, you can only do it with the API key. Would love to hear your thoughts, but this may be better served in a separate discussion or issue.
I use the REST API for some functionality that isn't in the SOAP API, and I'd be interested in contributing to the REST efforts.

**Comment by bigjohn98pl** [comment]
Hi Jesper,

I was using your lib in work for some time and I discover that there is still a lot of features that can be implemented, I learn a lot trying to automate importing and exporting data. I want to help develop this library. Unfortunately, I no longer have access to Polarion so I need to find out how I can get access to it. Can anyone know where and how I can set up my own Polarion project?

**Reply by jesper-raemaekers** [reply]
Shoot me an email at j.raemaekers@relek.nl and i'll help you get setup

**Comment by t3hcatpaw** [comment]
Hi Jesper,

I've been using your library extensively for work during the last year or so. In case you're still looking for maintainers, I would like to contribute to the project, dedicating maybe one or two hours per week.
That said, although I don't have more than superficial experience with SOAP/zeep (yet), I would be willing to enhance existing work. However, similar to what others mentioned before, I'd especially love to support in a potential migration towards REST.
Disclaimer: I've recently worked with REST interfacing MS DevOps, and for Polarion I've just succeeded implementing attachment download for documents, which is currently not otherwise implemented in this library (trying to re-use the code applied to work items ended in a zeep error, which I didn't spend time on to debug).
